toolchain/check/call.cpp

```diff
@@ -42,6 +63,12 @@ auto PerformCall(Context& context, Parse::NodeId node_id,
       return PerformCallToGenericClass(context, node_id,
                                        generic_class->class_id, arg_ids);
     }
+    if (auto generic_interface =
+            context.types().TryGetAs<SemIR::GenericInterfaceType>(
+                context.insts().Get(callee_id).type_id())) {
```

Comment:
Note here that you're doing twice (with generic_class) the equivalent of: `context.types().GetAsInst(context.insts().Get(callee_id).type_id())` -- one type lookup and two inst lookups. That seems worth refactoring to share, maybe use the CARBON_KIND_SWITCH?

Reply:
Done.